chore: reorder docs and rename Class based workflow to Model based workflow

abhilash-kumar-nair opened the pull request.

iakovn reviewed (changes requested):
Thanks for starting this! 🍑 

Please also rename "Class name" into "Model class name" in:
- `modelon/impact/client/entities/model.py`
- `modelon/impact/client/entities/interfaces/model.py`

abhilash-kumar-nair commented:
> Thanks for starting this! 🍑
> 
> Please also rename "Class name" into "Model class name" in:
> 
> * `modelon/impact/client/entities/model.py`
> * `modelon/impact/client/entities/interfaces/model.py`

Good catch! Fixed it! Thanks. :)

iakovn commented on docs/source/quick_start.rst:
```diff
@@ -18,27 +18,25 @@ A simple example using the :ref:`FMU based workflow <FMU based workflow>` is sho
    dynamic = workspace.get_custom_function('dynamic')
 
-   # Compile model
+   # Get model class
```
Get model reference. ("class" may be confusing)

iakovn commented on modelon/impact/client/entities/interfaces/model.py:
```diff
@@ -5,5 +5,5 @@ class ModelInterface(ABC):
     @abstractmethod
     def name(self) -> str:
-        """Class name."""
+        """Model class name."""
```
Sorry for confusion, drop the "class", just "Model name."

iakovn commented on modelon/impact/client/entities/model.py:
```diff
@@ -86,14 +86,14 @@ def __init__(
     @property
     def name(self) -> str:
-        """Class name."""
+        """Model class name."""
```
Model name.

iakovn reviewed (changes requested):
Minor tweaking and please update the tests.

abhilash-kumar-nair commented:
> Minor tweaking and please update the tests.

Fixed it now! Thanks!

iakovn commented on modelon/impact/client/entities/experiment.py:
```diff
@@ -541,7 +541,7 @@ def _get_workflow(self) -> _Workflow:
 
     def get_class_name(self) -> str:
-        """Return the model class name."""
+        """Return the model name."""
```
Given the name of the method the docstring would be unexpected (while `model class name` is ok-ish) . Shall we **add** `get_model_name` and eventually deprecate `get_class_name` (keep for awhile anyway)?

iakovn commented on modelon/impact/client/entities/model_executable.py:
```diff
@@ -253,7 +253,7 @@ def input(self) -> ModelExecutableInput:
 
     def get_class_name(self) -> str:
-        """Return the model class name."""
+        """Return the model name."""
```
same

iakovn reviewed (changes requested):
Unsure about `get_class_name` vs `get_model_name` for the method name.

abhilash-kumar-nair commented:
> Unsure about `get_class_name` vs `get_model_name` for the method name.

Fixed it! Thanks!

iakovn commented on modelon/impact/client/entities/model_executable.py:
```diff
@@ -123,7 +123,7 @@ def __init__(self, data: Dict[str, Any]):
     @property
     def class_name(self) -> str:
-        """Returns the model class name."""
+        """Returns the model name."""
```
Deprecate and add "model_name" property.

iakovn commented on modelon/impact/client/entities/model_executable.py:
```diff
@@ -253,7 +253,12 @@ def input(self) -> ModelExecutableInput:
 
     def get_class_name(self) -> str:
-        """Return the model class name."""
+        """Deprecated, use 'get_model_name' attribute."""
```
"""Deprecated, use 'get_model_name' method."""

iakovn commented on modelon/impact/client/entities/model_executable.py:
```diff
@@ -253,7 +253,12 @@ def input(self) -> ModelExecutableInput:
     def get_class_name(self) -> str:
-        """Return the model class name."""
+        """Deprecated, use 'get_model_name' attribute."""
+        logger.warning("This attribute is deprectated, use 'get_model_name' instead!")
```
How about :
```
import warnings
@warnings.deprecated("The 'get_class_name' is deprecated, use 'get_model_name'")
def get_class_name(self) -> str:
```
if you think this is too much then please fix the spelling of deprec**t**ated

iakovn reviewed (changes requested):
some more polishing

abhilash-kumar-nair commented:
> some more polishing

Good catch! :microscope: ! I have updated the code now.  there is no @warnings.deprecated unfortunately. I have switched to warnings.warn instead

iakovn reviewed (approved):
Looking good! ✈️
